src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ArrayConverter.cs

```diff
@@ -16,9 +17,9 @@ internal sealed class ArrayConverter<TCollection, TElement>
     {
         internal override bool CanHaveIdMetadata => false;
 
-        protected override void Add(TElement value, ref ReadStack state)
+        protected override void Add([AllowNull] TElement value, ref ReadStack state)
```

Comment:
A generic type when non-constrained to class or struct can be inferred as possible null already.

Reply:
True, `TElement` itself should determine whether null can be passed here. Will revert the change to these `Add` method signatures.